### PR TITLE
Add useful default titles to course guides

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**

### CODE REVIEW

File: app/lib/coursesHelper.coffee [hunk 1]
```diff
@@ -1,6 +1,21 @@
 Levels = require 'collections/Levels'
 utils = require 'core/utils'
 
+courseShortName = (course) ->
```
Comment (AndrewJakubowicz) [inline app/lib/coursesHelper.coffee]: Refactored this out from method below.

File: app/lib/coursesHelper.coffee [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+  switch
+    when /game-dev/.test(course.get('slug')) then 'GD'
+    when /web-dev/.test(course.get('slug')) then 'WD'
+    else 'CS'
+
+shortenCourseName = (course) ->
```
Comment (AndrewJakubowicz) [inline app/lib/coursesHelper.coffee]: The names usually have a number if it is needed except "Introduction to Computer Science".
Therefore if no number is found, assume it is the first, or 1.

File: app/lib/coursesHelper.coffee [hunk 3]
```diff
@@ -225,12 +240,11 @@ module.exports =
     labels = []
     courseLabelIndexes = CS: 0, GD: 0, WD: 0
     for course in courses
-      acronym = switch
-        when /game-dev/.test(course.get('slug')) then 'GD'
-        when /web-dev/.test(course.get('slug')) then 'WD'
-        else 'CS'
+      acronym = courseShortName(course)
```
Comment (AndrewJakubowicz) [inline app/lib/coursesHelper.coffee]: In the future could fully refactor this function to not store a counter.
No regression tests in place.

File: app/lib/coursesHelper.coffee (hunk 1 above)
Comment (nwinter) [reply app/lib/coursesHelper.coffee]: We actually have this refactored out in the [Course model](https://github.com/codecombat/codecombat/tree/master/app/models/Course.coffee#L17-L25), too. Should probably get it down to just one.

File: app/lib/coursesHelper.coffee (hunk 2 above)
Comment (nwinter) [reply app/lib/coursesHelper.coffee]: These two method names are confusing. Neither of them return a short name; really, they return an acronym, and one includes the number and the other doesn't. One is a noun and the other a verb...

File: app/lib/coursesHelper.coffee (hunk 1 above)
Comment (AndrewJakubowicz) [reply app/lib/coursesHelper.coffee]: Thank you for pointing this out to me.

File: app/models/Course.coffee [hunk 4]
```diff
@@ -15,11 +15,10 @@ module.exports = class Course extends CocoModel
...
 
   acronym: ->
-    # TODO: refactor coursesHelper coursesLabelsArray
     # TODO: i18n (optional parameter so we can still get English acronym, too)
     acronym = switch
       when /game-dev/.test(@get('slug')) then 'GD'
       when /web-dev/.test(@get('slug')) then 'WD'
       else 'CS'
-    number = @get('slug').match(/(\d+)$/)?[1] or '1'
+    number = @get('slug')?.match(/(\d+)$/)?[1] or '1'
```
Comment (AndrewJakubowicz) [inline app/models/Course.coffee]: It's possible for 'slug' to be undefined before the model is populated.

**Pull request merged**